### PR TITLE
feat(styles): add disabled styles for SfButton

### LINKED ISSUE #603: [FEAT] Add disabled styles to SfButton
Please add disabled styles to SfButton
[[Design]](https://www.figma.com/file/hrwE3VsMBHgdJoS86rVr4W/Desktop-%26-Mobile-Vue-Storefront?node-id=454%3A516)
Zrzut ekranu 2019-12-27 o 12 13 33

### PR DESCRIPTION
**Pull request by aniamusial**
# Related issue
Closes #603 

# Scope of work
Added --disabled modifier for SfButton

# Checklist

- [x] I followed [composition rules](https://docs.storefrontui.io/contributing/coding-guidelines.html#component-rules) for my component
- [x] I tested the component in most common device sizes (can be tested in Storybook [from viewport addon in top menu](https://github.com/storybooks/storybook/tree/master/addons/viewport))

### CODE REVIEW

File: packages/shared/styles/components/SfButton.scss [hunk 1]
```diff
@@ -76,4 +76,14 @@ $button--outline-border: 2px !default;
       }
     }
   }
+
+  &--disabled {
```
Comment (pspaczek) [inline packages/shared/styles/components/SfButton.scss]: @aniamusial it should be work when button is disabled :)

**Comment by pspaczek** [comment]
awesome, @aniamusial can you add transition when you remove and add `disabled`

**Comment by aniamusial** [comment]
@pspaczek added transition :)

File: packages/shared/styles/components/SfButton.scss [hunk 2]
```diff
@@ -21,6 +21,7 @@ $button--outline-border: 2px !default;
   font-size: $button-font-size;
   font-weight: $button-font-weight;
   line-height: $button-line-height;
+  transition: background-color 0.3s ease;
```
Comment (pspaczek) [inline packages/shared/styles/components/SfButton.scss]:
```suggestion
  transition: background-color 150ms linear, color 150ms linear;
```

File: packages/shared/styles/components/SfButton.scss [hunk 3]
```diff
@@ -76,4 +77,15 @@ $button--outline-border: 2px !default;
       }
     }
   }
+
+  &:disabled {
+    background-color: $c-light;
+    color: $c-gray;
+    cursor: default;
+    transition: background-color 0.3s ease;
```
Comment (pspaczek) [inline packages/shared/styles/components/SfButton.scss]:
```suggestion
    transition: background-color 150ms linear, color 150ms linear;
```

**Comment by aniamusial** [comment]
@pspaczek looks good to me

**Pull request merged**